src/sync/impl/sync_metadata.hpp

```diff
@@ -188,12 +207,9 @@ friend class SyncFileActionMetadata;
 
 private:
     SyncUserMetadataResults get_users(bool marked) const;
-
     Realm::Config m_metadata_config;
-
     SyncUserMetadata::Schema m_user_schema;
     SyncFileActionMetadata::Schema m_file_action_schema;
-    mutable std::mutex m_metadata_lock;
```

Comment:
I actually don't think this was necessary in the first place, since the config was created in the constructor and only ever read by external consumers. It's definitely not necessary anymore, since the config is created in the constructor, never changed, and only used by code internal to this class.